### ISSUE TITLE
Can't add label without format key=value to container

### ISSUE THREAD

**Issue by rreinurm** [moby/moby#36497]
**Description**
Since version 18.03.0-ce-rc1, its not possible to add labels to containers just specifying key-only meaning with empty string as described here https://docs.docker.com/engine/reference/commandline/run/#set-metadata-on-container--l-label-label-file

I think issue was introduced by
https://github.com/docker/cli/pull/838
because validation function https://github.com/docker/cli/blob/2bfac7fcdafeafbd2f450abb6d1bb3106e4f3ccb/opts/opts.go#L268-L275 checks presence of  `=`

**Steps to reproduce the issue:**
1. docker run --rm --label mylabel alpine true

**Describe the results you received:**
invalid argument "mylabel" for "-l, --label" flag: bad attribute format: mylabel

**Describe the results you expected:**
Container should be created with provided label key as mylabel and value as empty string

**Output of `docker version`:**

```
Client:
 Version:	18.03.0-ce-rc1
 API version:	1.37
 Go version:	go1.9.4
 Git commit:	c160c73
 Built:	Thu Feb 22 02:34:03 2018
 OS/Arch:	darwin/amd64
 Experimental:	false
 Orchestrator:	swarm

Server:
 Engine:
  Version:	18.03.0-ce-rc1
  API version:	1.37 (minimum version 1.12)
  Go version:	go1.9.4
  Git commit:	c160c73
  Built:	Thu Feb 22 02:42:37 2018
  OS/Arch:	linux/amd64
  Experimental:	true
```

**Output of `docker info`:**

```
ontainers: 6
 Running: 4
 Paused: 0
 Stopped: 2
Images: 8
Server Version: 18.03.0-ce-rc1
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host ipvlan macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: cfd04396dc68220d1cecbe686a6cc3aa5ce3667c
runc version: 6c55f98695e902427906eed2c799e566e3d3dfb5
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.75-linuxkit-aufs
Operating System: Docker for Mac
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 1.952GiB
Name: linuxkit-025000000001
ID: CQKC:26XS:WAJZ:CGOV:VLVH:QB23:LCGD:6OCQ:77NQ:FDZI:ZG36:LMQ4
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 47
 Goroutines: 61
 System Time: 2018-03-06T12:41:31.913991655Z
 EventsListeners: 3
HTTP Proxy: docker.for.mac.http.internal:3128
HTTPS Proxy: docker.for.mac.http.internal:3129
Registry: https://index.docker.io/v1/
Labels:
Experimental: true
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Docker-for-Mac

**Comment by thaJeztah** [comment]
Thanks for reporting; this issue has been fixed on master through https://github.com/docker/cli/pull/904, and will be included in Docker CE 18.03.0-rc2 through https://github.com/docker/docker-ce/pull/447

**Comment by thaJeztah** [comment]
I'll close this issue, because it has been fixed, but feel free to continue the conversation, and thanks again for reporting 👍
